src/Countries/Singapore.php

```diff
@@ -0,0 +1,171 @@
+<?php
+
+namespace Spatie\Holidays\Countries;
+
+use Carbon\CarbonImmutable;
+use DateTime;
+use DateTimeZone;
+use IntlDateFormatter;
+use Solaris\MoonPhase;
+
+class Singapore extends Country
+{
+
+    protected string $timezone = "Asia/Singapore";
+
+    public function countryCode(): string
+    {
+        return 'sg';
+    }
+
+    protected function allHolidays(int $year): array
+    {
+        return array_merge(
+            $this->fixedHolidays($year),
+            $this->variableHolidays($year)
+        );
+    }
+
+    /** @return array<string, string|CarbonImmutable> */
+    protected function fixedHolidays(int $year): array
+    {
+        #Fixed holidays in Singapore
+        return array_merge(
+            $this->replacementForSunday("New Year's Day", $year, "01-01", 1),
+            $this->replacementForSunday("Labor Day", $year, "05-01", 1),
+            $this->replacementForSunday("National Day", $year, "08-09", 1),
+            $this->replacementForSunday("Christmas Day", $year, "12-25", 1),
+        );
+    }
+
+    /** @return array<string, string|CarbonImmutable> */
+    protected function variableHolidays(int $year): array
+    {
+
+        $easter = $this->easter($year);
+
+        return array_merge(
+            $this->replacementForSunday("Chinese New Year Day 1", $year, $this->chineseCalendar("01-01", $year), 2),
+            $this->replacementForSunday("Chinese New Year Day 2", $year, $this->chineseCalendar("01-02", $year), 1),
+            $this->replacementForSunday("Hari Raya Pausa", $year, $this->islamicCalendar("01-10", $year), 1),
+            $this->replacementForSunday("Vesak Day", $year, $this->chineseCalendar("04-15", $year), 1),
+            $this->replacementForSunday("Hari Raya Haji", $year, $this->islamicCalendar("10-12", $year), 1),
+            $this->getDeepavali($year),
+            [
+                'Good Friday' => $easter->subDays(2),
+            ]
+        );
+    }
+
+    /** @return array<string, string|CarbonImmutable> */
+    protected function replacementForSunday(string $name, int $year, string $input, int $addDays = 0): array
```

Comment:
Could you make this name & code more readable, so it's immediately obvious what it does?

Reply:
I will review on this.